static/js/filter.js

```diff
@@ -108,7 +119,7 @@ function message_matches_search_term(message, operator, operand) {
         if (!operand_ids) {
             return false;
         }
-        var message_ids = people.pm_with_user_ids(message);
+        message_ids = people.pm_with_user_ids(message);
```

Comment:
I see this is where you got the `message_ids` name from.  I'll merge a quick commit fixing that name.